File: solutions/3254. Find the Power of K-Size Subarrays I/3254.py
# Time complexity: O(n)
# Space complexity: O(n)
class Solution:
  def resultsArray(self, nums: List[int], k: int) -> List[int]:
    ans = []
    start = 0

    for i, num in enumerate(nums):
      if i > 0 and num != nums[i - 1] + 1:
        start = i
      if i >= k - 1:
        ans.append(num if i - start + 1 >= k else -1)

    return ans

File: solutions/3254. Find the Power of K-Size Subarrays I/3254.rs
// Time complexity: O(n)
// Space complexity: O(n)
impl Solution {
    pub fn results_array(nums: Vec<i32>, k: i32) -> Vec<i32> {
        let k = k as usize;
        let mut ans = Vec::new();
        let mut start = 0;

        for i in 0..nums.len() {
            if i > 0 && nums[i] != nums[i - 1] + 1 {
                start = i;
            }
            if i >= k - 1 {
                if i - start + 1 >= k {
                    ans.push(nums[i]);
                } else {
                    ans.push(-1);
                }
            }
        }

        ans
    }
}
